flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/SpilledSubpartitionView.java

```diff
@@ -219,6 +219,14 @@ public boolean nextBufferIsEvent() {
 		}
 	}
 
+	@Override
+	public synchronized boolean isAvailable() {
+		if (nextBuffer != null) {
+			return true;
+		}
+		return fileReader.hasReachedEndOfFile();
```

Comment:
should be `!fileReader.hasReachedEndOfFile();` - I will add tests in a later PR for this